File: okhttpprofiler/src/main/java/com/localebro/okhttpprofiler/transfer/DataTransfer.java
package com.localebro.okhttpprofiler.transfer;

import java.io.IOException;

import okhttp3.Request;
import okhttp3.Response;

public interface DataTransfer {
    void sendRequest(String id, Request request) throws IOException;

    void sendResponse(String id, Response response) throws IOException;

    void sendException(String id, Exception response);

    void sendDuration(String id, long duration);
}

File: okhttp-profiler/src/main/java/io/nerdythings/okhttp/profiler/transfer/DataTransfer.kt
package io.nerdythings.okhttp.profiler.transfer

import okhttp3.Request
import okhttp3.Response
import java.io.IOException

interface DataTransfer {
    @Throws(IOException::class)
    fun sendRequest(id: String, request: Request)

    @Throws(IOException::class)
    fun sendResponse(id: String, response: Response)

    fun sendException(id: String, response: Exception)

    fun sendDuration(id: String, duration: Long)
}
